File: app/src/main/java/com/suhane/flickr/di/modules/ActivityModule.java
package com.suhane.flickr.di.modules;

import android.app.Activity;

import com.suhane.flickr.di.scopes.ActivityScope;

import dagger.Module;
import dagger.Provides;

/**
 * Created by shashanksuhane on 04/02/18.
 */

@Module
public class ActivityModule {
    private final Activity activityContext;

    public ActivityModule(Activity activityContext) {
        this.activityContext = activityContext;
    }

    @Provides
    @ActivityScope
    Activity getActivityContext() {
        return activityContext;
    }
}

File: app/src/main/java/com/suhane/flickr/di/modules/ActivityModule.kt
package com.suhane.flickr.di.modules

import android.app.Activity

import com.suhane.flickr.di.scopes.ActivityScope

import dagger.Module
import dagger.Provides

/**
 * Created by shashanksuhane on 04/02/18.
 */

@Module
class ActivityModule(private val activityContext: Activity) {
    @Provides
    @ActivityScope
    fun getActivityContext(): Activity {
        return activityContext
    }
}
